### PR TITLE
delete fix for dataadapter.

### PR DESCRIPTION
**Pull request by benwulfe**
More testing for DataAdapter revealed that delete had an error.
More tests coming, but would like to include this fix in the alpha update tonight.

### CODE REVIEW

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs [hunk 1]
```diff
@@ -155,6 +161,11 @@ IDbCommand IDbDataAdapter.UpdateCommand
         }
 
         /// <summary>
+        /// The set of primary keys defined for <see cref="AutoGeneratedCommandTable"/>.
+        /// </summary>
+        public HashSet<string> AutoGeneratedCommandPrimaryKeys { get; } = new HashSet<string>();
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs]: Does this need to be exposed as a mutable collection? Or exposed at all?

Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs]: yes for the case where the developer uses the default ctor (which is the designer scenario).
Not sure if we'll be able to get the toolbox working near term, but I don't want to cut off the use case.

Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs]: The user does not need to know it is a `HashSet<string>`. The `ISet<string>` interface would do.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs [hunk 2]
```diff
@@ -119,11 +120,16 @@ public string AutoGeneratedCommandTable
...
+        /// <param name="primaryKeys">The set of columns that form the primary key for <paramref name="autoGeneratedCommandTable"/>.</param>
+        public SpannerDataAdapter(SpannerConnection connection, string autoGeneratedCommandTable, params string[] primaryKeys)
         {
             GaxPreconditions.CheckNotNull(connection, nameof(connection));
             SpannerConnection = connection;
             AutoGeneratedCommandTable = autoGeneratedCommandTable;
+            if (primaryKeys != null)
+            {
+                AutoGeneratedCommandPrimaryKeys.UnionWith(primaryKeys);
+            }
```
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs]: It is just a matter of preference, but I tend to think lines like this look better with null coalescing.
```
AutoGeneratedCommandPrimaryKeys.UnionWith(primaryKeys ?? Enumerable.Empty<string>());
```

Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataAdapter.cs]: interesting option.  next time :)  thanks for the suggestion.

**Pull request merged**